components/brave_wallet/renderer/js_ethereum_provider.cc

```diff
@@ -200,13 +196,25 @@ void JSEthereumProvider::Install(bool allow_overwrite_window_ethereum_provider,
     return;
   }
 
-  if (!allow_overwrite_window_ethereum_provider) {
+  // Set window.ethereumProvider
+  {
     SetProviderNonWritable(context, global, ethereum_proxy,
-                           gin::StringToV8(isolate, kEthereum), true);
-  } else {
-    global
-        ->Set(context, gin::StringToSymbol(isolate, kEthereum), ethereum_proxy)
-        .Check();
+                           gin::StringToV8(isolate, kBraveEthereum), true);
+  }
+
+  v8::Local<v8::Value> ethereum_value =
+      global->Get(context, gin::StringToV8(isolate, kEthereum))
+          .ToLocalChecked();
+  if (install_ethereum_provider && ethereum_value->IsUndefined()) {
```

Comment:
this check is completely unnecessary,  please keep it as what it was.

Reply:
If you are talking about checking whether ethereum_value->IsUndefined() - it is taken from line 170

Reply:
I was referring to `install_ethereum_provider` and also we should check `kBraveEthereum` instead as I mentioned in https://github.com/brave/brave-core/pull/18751#discussion_r1269683945 because `window.braveEthereum` and its alias `window.ethereum` will be installed together.